src/features/Events.js

```diff
@@ -0,0 +1,184 @@
+import React, { Fragment, useState } from 'react';
+import { useSelector } from 'react-redux';
+import dayjs from 'dayjs';
+import { orderBy } from 'lodash';
+import cx from 'classnames';
+import bootstrap from 'bootstrap/dist/css/bootstrap.css';
+
+import {
+  selectHours,
+  selectSelectedCalendar,
+  selectEventsByRange,
+  selectDate,
+} from '../stores/viewState';
+import { selectCalendars } from '../stores/calendars';
+import createBlobUrl from '../utils/createBlobUrl';
+import { SORT_BY } from '../constants';
+
+import styles from './Events.module.css';
+
+const Events = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [sortBy, setSortBy] = useState(SORT_BY.DATE);
+
+  const selectedCalendar = useSelector(selectSelectedCalendar);
+  const events = useSelector((state) =>
+    selectEventsByRange(state, selectedCalendar)
+  );
+  const calendars = useSelector(selectCalendars);
+  const date = useSelector(selectDate);
+
+  const rangeHours = useSelector(selectHours);
+
+  if (!rangeHours) {
+    return null;
+  }
+
+  const currentCalendarName = calendars.find(
+    (item) => item.id === selectedCalendar
+  )?.label;
+
+  let eventsToRender = events.map((event) => {
+    const itemDateStart = new Date(event.start.dateTime);
+    const itemDateEnd = new Date(event.end.dateTime);
+
+    const hours = (itemDateEnd - itemDateStart) / 1000 / 60 / 60;
+
+    return {
+      ...event,
+      hours,
+    };
+  });
+
+  let downloadBlob;
+  let filename;
+
+  if (sortBy === SORT_BY.AMOUNT) {
+    const eventsObject = {};
+    eventsToRender.forEach(({ summary, hours }) => {
+      if (eventsObject[summary]) {
+        eventsObject[summary] += hours;
+      } else {
+        eventsObject[summary] = hours;
+      }
+    });
+    const newArray = Object.entries(eventsObject).map(([key, value]) => ({
+      summary: key,
+      hours: value,
+      id: key,
+    }));
+    eventsToRender = orderBy(newArray, 'hours', 'desc');
+  } else {
+    const lines = eventsToRender.map(
+      ({ start, end, summary, hours }) =>
+        `${dayjs(start.dateTime).format('DD.MM.YYYY HH:mm')},${dayjs(
+          end.dateTime
+        ).format('DD.MM.YYYY HH:mm')},"${summary}",${hours}`
```

Comment:
we should think about making this configurable. not everyone is used to `DD.MM.YYYY HH:mm`

Reply:
and maybe put this in the constants as well?

Reply:
There is already an issue for this: https://github.com/aronwoost/google-calendar-hours/issues/66 😊

I moved it to a CONST for now.